pjsip/src/pjsua-lib/pjsua_call.c

```diff
@@ -5576,13 +5576,18 @@ static void pjsua_call_on_tsx_state_changed(pjsip_inv_session *inv,
 		found_idx = pj_strtok(&input, &delim, &token, 0);
 		if (found_idx != input.slen) {
 		    /* Get signal/digit */
-		    const pj_str_t STR_SIGNAL = { "Signal=", 7 };
-		    const pj_str_t STR_DURATION = { "Duration=", 9 };
+		    const pj_str_t STR_SIGNAL = { "Signal", 6 };
+		    const pj_str_t STR_DURATION = { "Duration", 8 };
 		    char *val;
 
 		    val = pj_strstr(&input, &STR_SIGNAL);
 		    if (val) {
-			info.digit = *(val+STR_SIGNAL.slen);
+				pj_ssize_t idx1 = val - input.ptr;
+				pj_ssize_t idx2 = 0;
+				while ( (*(val + STR_SIGNAL.slen + idx2) == 32 || *(val + STR_SIGNAL.slen + idx2) == 61) &&
+					(idx1 + idx2 < input.slen))
+					idx2++;
+			info.digit = *(val+STR_SIGNAL.slen + idx2);
```

Comment:
what if there is no `=` in the input string?

Reply:
I have thought of this case. But I don't know what to return in case of error format.

Reply:
In case of error format, I think we should not call the `on_dtmf_digit2()` callback (and leave `is_handled` to false so 400 response will be sent) and write some log info?

> I have thought of this case. But I don't know what to return in case of error format.